src/descriptor/key.rs

```diff
@@ -112,7 +125,7 @@ pub struct DescriptorXKey<K: InnerXKey> {
     /// The derivation path
     pub derivation_path: bip32::DerivationPath,
     /// Whether the descriptor is wildcard
-    pub is_wildcard: bool,
+    pub is_wildcard: Wildcard,
```

Comment:
nit: Change to `wildcard` instead of `is_wildcard` since this is no longer a bool?